### PR TITLE
A computed property key must be a string

### PR DESCRIPTION
**Pull request by GavinJoyce**
https://embercommunity.slack.com/archives/dev-ember/p1454255249000443

image

### CODE REVIEW

File: packages/ember-metal/lib/expand_properties.js [hunk 1]
```diff
@@ -37,26 +36,26 @@ var END_WITH_EACH_REGEX = /\.@each$/;
   expansion, and is passed the expansion.
 */
 export default function expandProperties(pattern, callback) {
+  if (typeof pattern !== 'string') {
+    throw new EmberError('A computed property key must be a string');
```
Comment (rwjblue) [inline packages/ember-metal/lib/expand_properties.js]: Can we change this to be an `assert` instead?  I'd hate to continue doing this `indexOf` in prod builds just to provide a more helpful message than the error that would already be thrown...

File: packages/ember-metal/lib/expand_properties.js [hunk 2]
```diff
@@ -37,26 +36,26 @@ var END_WITH_EACH_REGEX = /\.@each$/;
...
-    var properties = [parts];
+  var parts = pattern.split(SPLIT_REGEX);
+  var properties = [parts];
 
-    parts.forEach((part, index) => {
-      if (part.indexOf(',') >= 0) {
-        properties = duplicateAndReplace(properties, part.split(','), index);
-      }
-    });
+  parts.forEach((part, index) => {
```
Comment (rwjblue) [inline packages/ember-metal/lib/expand_properties.js]: I know this seems completely unrelated to the point of this PR, but could you use a "normal" for loop here instead?

File: packages/ember-metal/lib/expand_properties.js [hunk 3]
```diff
@@ -37,26 +36,26 @@ var END_WITH_EACH_REGEX = /\.@each$/;
...
+    }
+  });
 
-    properties.forEach((property) => {
-      callback(property.join('').replace(END_WITH_EACH_REGEX, '.[]'));
-    });
-  } else {
-    callback(pattern.replace(END_WITH_EACH_REGEX, '.[]'));
-  }
+  properties.forEach((property) => {
```
Comment (rwjblue) [inline packages/ember-metal/lib/expand_properties.js]: Same as above (swap from `.forEach` to a for loop)...

**Comment by GavinJoyce** [comment]
@rwjblue thanks, updated.

**Comment by rwjblue** [comment]
There are a couple failures from preexisting tests (that were testing the prior throwing behavior).

**Comment by GavinJoyce** [comment]
Thanks @rwjblue, TIL about `expectAssertion`. Time to read [`CONTRIBUTING.md`](https://github.com/emberjs/ember.js/blob/master/CONTRIBUTING.md) again.

I've updated the failing tests, everything is green. Does the commit message need a prefix?

**Comment by rwjblue** [comment]
@GavinJoyce - Lets go with `[BUGFIX beta]` on this.

**Comment by GavinJoyce** [comment]
done and green

**Comment by rwjblue** [comment]
Thanks @GavinJoyce!

**Pull request merged**